### ISSUE TITLE
Checkout of follow-up draft change selects parent one

### ISSUE THREAD

**Issue by koral--** [uwolfer/gerrit-intellij-plugin#230]
Steps to reproduce:
1. Create change (non-draft, not submitted)
2. Create follow-up draft change (not sure if draft here matters or not)
3. Use checkout context menu options on the 2nd change
4. First one (parent) is checked out instead

Works fine with command-line `git-review -d <draft change #>` so it seems that is Intellij plugin issue.

Versions:
Android Studio 2.1.2 Build #AI-143.2915827
Plugin 0.9.8.1-142
Gerrit 2.12.2-1957-gc7cb3dd

**Comment by uwolfer** [comment]
@koral--: You can see git commands in the tab "Version control" -> "Console". Can you please check if correct commands are executed?

**Comment by ar-turek** [comment]
I have a similar issue, without the draft thing.

When I create a change in Gerrit and then use the plugin to checkout, a new branch gets created (review/artur_ciesielski/2806). This is fine.

But then pushing a new patch set and checking out again gives me the message (after fetching the change to FETCH_HEAD):
`git -c core.quotepath=false checkout -b review/artur_ciesielski/2806 FETCH_HEAD --`
`fatal: A branch named 'review/artur_ciesielski/2806' already exists.`
After which the plugin checks out the old (locally stored) branch (checked via Version Control console):
`git -c core.quotepath=false checkout review/artur_ciesielski/2806 --`

So basically I can't checkout a change without removing the local branch first.

**Comment by uwolfer** [comment]
@artur-ciesielski: This issue is related, but can be fixed independently. Can you please open a new issue for it?

**Comment by ar-turek** [comment]
Sure, on it.

**Comment by koral--** [comment]
@uwolfer here is the console content (sensitive information stripped):

```
18:44:23.246: cd <path>
18:44:23.246: git -c core.quotepath=false fetch gerrit refs/changes/20/5320/2 --progress --prune
From ssh://<host>/<project>
 * branch            refs/changes/20/5320/2 -> FETCH_HEAD
18:44:24.175: cd <path>
18:44:24.175: git -c core.quotepath=false checkout -b review/<user>/5320 FETCH_HEAD
fatal: A branch named 'review/<user>/5320' already exists.
18:44:24.194: cd <path>
18:44:24.195: git -c core.quotepath=false checkout review/<user>/5320
Switched to branch 'review/<user>/5320'
```

After that actions working directory does not contain file added in patchset 2 in change 5320.
Important update to the 1st post: it seems that instead of checking out parent change, nothing happens, working directory is left untouched in state before check out.
When writing 1st post I have just working directory containing exactly state from parent change so that was the same cases.
`git review -d 5320` produces correct state.

Not sure if it matters but branch name produced by Intellij plugin contains `ó` character while `git-review` replaces it by underscore (`_`).

**Comment by uwolfer** [comment]
@koral--: When you create local branches, do you use the same naming as used by the plugin (e.g. `review/<user>/5320`)? I'm asking this because the local-branch-create command fails with the error that the branch already exists.

**Comment by koral--** [comment]
I haven't created local branches in this project manually only ones are created by the Intellij plugin and `git-review` command-line tool but those don't clash due to character substitution.
On the other hand I have previously checked out patchset 1 of change 5320 using Intellij plugin that's why branch `review/<user>/5320` already exists. I don't remember what was the history when writing 1st post.

From pasted log I guess that plugin tries to create branch `review/<user>/5320` for patchset 2 but fails because such branch already exists and that error was effectively swallowed in UI (green tooltip appeared). Not sure how to fix this, maybe create branch name containing patchset number? `git-review` just does hard reset AFAIK: https://storyboard.openstack.org/#!/story/1096057

Second bug is that success is shown in UI. No matter how the core issue will be fixed, error should be shown in UI instead of success in cases when git returned an error (eg. someone manually created branch with the same name).

**Comment by uwolfer** [comment]
@koral--: Thanks for your detailed analysis.

It used to do `git reset`, but this was reported as "error" by many users (#205).

Can you also reproduce this issue when you checkout a change which you have not checked out before (and thus no local branch with same name exists)?

**Comment by ar-turek** [comment]
@uwolfer @koral-- I second the motion for displaying failure instead of success. This kind of bugs me as well.

**Comment by koral--** [comment]
Error is not reproducible when there was no checkout done previously.
IMHO reseting is also not good idea because local changes may be lost and behavior of git-review in that matter is a bug.

**Comment by eekboom** [comment]
Related/duplicate: #235, #267
